ref/Microsoft.Build.Framework/net/Microsoft.Build.Framework.cs

```diff
@@ -242,6 +242,22 @@ public partial interface IEventSource3 : Microsoft.Build.Framework.IEventSource,
         void IncludeEvaluationProfiles();
         void IncludeTaskInputs();
     }
+    public partial interface IFileSystem
+    {
+        void ClearCaches();
+        bool DirectoryEntryExists(string path);
+        bool DirectoryExists(string path);
+        System.Collections.Generic.IEnumerable<string> EnumerateDirectories(string path, string searchPattern="*", System.IO.SearchOption searchOption=(System.IO.SearchOption)(0));
+        System.Collections.Generic.IEnumerable<string> EnumerateFiles(string path, string searchPattern="*", System.IO.SearchOption searchOption=(System.IO.SearchOption)(0));
+        System.Collections.Generic.IEnumerable<string> EnumerateFileSystemEntries(string path, string searchPattern="*", System.IO.SearchOption searchOption=(System.IO.SearchOption)(0));
+        bool FileExists(string path);
+        System.IO.FileAttributes GetAttributesNoMissingException(string path);
+        System.IO.Stream GetFileStream(string path, System.IO.FileMode mode, System.IO.FileAccess access, System.IO.FileShare share);
+        System.IO.TextReader ReadFile(string path);
+        byte[] ReadFileAllBytes(string path);
```

Comment:
This doesn't seem like something MSBuild needs?

Reply:
But it does seem useful for sdk resolvers and cache plugins which read files.